spec/rubocop/cop/layout/align_hash_spec.rb

```diff
@@ -319,6 +319,16 @@ def example
       RUBY
     end
 
+    it 'registers an offense for misaligned hash keys with mixed hash style' do
+      expect_offense(<<~RUBY)
+        headers = {
+          "Content-Type" => 0,
+           Authorization: 1
+           ^^^^^^^^^^^^^^^^ Align the elements of a hash literal if they span more than one line.
+        }
+      RUBY
+    end
```

Comment:
`nil can't be coerced into Integer` does not reproduce when reverting product code.

```diff
diff --git a/lib/rubocop/cop/mixin/hash_alignment.rb b/lib/rubocop/cop/mixin/hash_alignment.rb
index 59d078c55..228bb73d6 100644
--- a/lib/rubocop/cop/mixin/hash_alignment.rb
+++ b/lib/rubocop/cop/mixin/hash_alignment.rb
@@ -102,7 +102,7 @@ module RuboCop
         end

         def hash_rocket_delta(first_pair, current_pair)
-          first_pair.loc.column + max_key_width.to_i + 1 -
+          first_pair.loc.column + max_key_width + 1 -
             current_pair.loc.operator.column
         end

@@ -111,7 +111,7 @@ module RuboCop

           correct_value_column = first_pair.key.loc.column +
                                  current_pair.delimiter(true).length +
-                                 max_key_width.to_i
+                                 max_key_width
           correct_value_column - current_pair.value.loc.column
         end
       end
```

```console
% bundle exec rspec spec/rubocop/cop/layout/align_hash_spec.rb
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 52306
......................................................................................

Finished in 0.19007 seconds (files took 2 seconds to load)
86 examples, 0 failures

Randomized with seed 52306
```

Test cases for the following .rubocop.yml setting need to be written.

```yaml
Layout/AlignHash:
  EnforcedHashRocketStyle:
    - key
    - table
```

The existing `let(:cop_config)` may be equivalent to that.

```ruby
     let(:cop_config) do
       {
         'EnforcedHashRocketStyle' => %w[key table],
```

Reply:
Yes, you are absolutely correct. I originally wrote those specs in the `%w[key table]` section when I wrote the fix, but then moved them thinking it was more of a generic problem. That was silly of me.

Fixed and force-pushed to correct it.